lib/Rx/Subject/ReplaySubject.php

```diff
@@ -0,0 +1,171 @@
+    {
+        if ($bufferSize === null || !is_int($bufferSize)) {
+            $bufferSize = $this->maxSafeInt;
+        }
+        if ($bufferSize >= 0) {
+            $this->bufferSize = $bufferSize;
+        }
+
+        if ($windowSize === null || !is_int($windowSize)) {
+            $windowSize = $this->maxSafeInt;
+        }
+        if ($windowSize >= 0) {
+            $this->windowSize = $windowSize;
+        }
+
+        if ($scheduler instanceof SchedulerInterface) {
+            $this->scheduler = $scheduler;
+        }
+    }
+
+    public function subscribe(ObserverInterface $observer, $scheduler = null)
+    {
+        $this->assertNotDisposed();
+
+        if (!($scheduler instanceof SchedulerInterface)) {
+            $scheduler = $this->scheduler;
+        }
+        $so = new ScheduledObserver($scheduler, $observer);
+
+        $subscription = $this->createRemovableDisposable($this, $so);
+
+        $this->trim();
+
+        $this->observers[] = $so;
+
+        foreach ($this->queue as $item) {
+            $so->onNext($item["value"]);
+        }
+
+        if ($this->hasError) {
+            $so->onError($this->exception);
+        } else {
+            if ($this->isStopped) {
+                $so->onCompleted();
+            }
+        }
+
+        $so->ensureActive();
+
+        return $subscription;
+    }
+
+    public function onNext($value)
+    {
+        $now = 0;
+        if ($this->scheduler instanceof SchedulerInterface) {
+            $now = $this->scheduler->now();
+        }
+        $this->queue[] = ["interval" => $now, "value" => $value];
+        $this->trim();
+
+        $ret = parent::onNext($value);
+
+        /** @var ScheduledObserver $observer */
+        foreach ($this->observers as $observer) {
+            $observer->ensureActive();
+        }
+
+        return $ret;
+    }
+
+    public function onCompleted()
+    {
+        $this->assertNotDisposed();
+
+        if ($this->isStopped) {
+            return;
+        }
+
+        $observers       = $this->observers;
+        $this->isStopped = true;
+
+        foreach ($observers as $observer) {
+            $observer->onCompleted();
+            $observer->ensureActive();
+        }
+
+        $this->observers = [];
+    }
+
+    public function onError(Exception $exception)
+    {
+        $this->assertNotDisposed();
+
+        if ($this->isStopped) {
+            return;
+        }
+
+        $observers       = $this->observers;
+        $this->isStopped = true;
+        $this->exception = $exception;
+        $this->hasError  = true;
+
+        $this->trim();
+
+        foreach ($observers as $observer) {
+            $observer->onError($exception);
+            $observer->ensureActive();
+        }
+
+        $this->observers = [];
+    }
+
+
+    private function createRemovableDisposable($subject, $observer)
+    {
+        return new CallbackDisposable(function () use ($observer, $subject) {
+            $observer->dispose();
```

Comment:
`$observer` has no `dispose` method here? (Also add typehints to `createRemovableDisposable()`.)